### PR TITLE
Task_1_4_1

### PR DESCRIPTION
**Pull request by cookie0629**

### CODE REVIEW

File: Task_1_4_1/src/main/java/ru/nsu/zhao/FITBook.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    }
+
+    /**
+     * 设置最终考试成绩。
+     *
+     * @param score 考试成绩
+     */
+    public void setExamScore(FIT score) {
+        this.examScore = score;
+        this.examFlag = true;
```
Comment (es-cheryomushkin) [inline Task_1_4_1/src/main/java/ru/nsu/zhao/FITBook.java:45]: Dear Zhao Sun Thao,
Your code looks clean and good. If you don't mind I will make some minor notes, that may help you in the future.

Variable names should be self explanatory. For example here you can call the variable more precisely examFlag -> examPassed - or what does that variable means. I don't know for sure what is the meaning of this variable.

Please feel free to aske any question regarding the code.

File: Task_1_4_1/src/main/java/ru/nsu/zhao/FITBook.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * 学生成绩记录簿类，包含学期成绩、考试成绩以及是否为公费等信息。
+ */
+public class FITBook {
+    private final List<List<FITRecord>> semesters; // 所有学期的成绩记录
```
Comment (es-cheryomushkin) [inline Task_1_4_1/src/main/java/ru/nsu/zhao/FITBook.java:12]: 1. Looks like some variables from this list can be final. Please put final where possible.
2. comments should be in javadoc style everywhere please correct.

File: Task_1_4_1/src/main/java/ru/nsu/zhao/FITBook.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+     *
+     * @return 每个学科的最新成绩的映射
+     */
+    private Map<String, FIT> getLastGradeForEachSubject() {
+        return semesters.stream() // 创建学期的流
+                .flatMap(List::stream) // 将所有学期的记录展平为单个流
+                .collect(Collectors.toMap( // 收集流中的元素到 Map
+                        FITRecord::getSubjectName, // 键为科目名称
+                        FITRecord::getScore, // 值为对应的成绩
+                        (existing, replacement) -> replacement // 如果有重复科目，保留最后一次的成绩
```
Comment (es-cheryomushkin) [inline Task_1_4_1/src/main/java/ru/nsu/zhao/FITBook.java:75]: This is good idea to use collectors.toMap here

File: Task_1_4_1/src/main/java/ru/nsu/zhao/FITBook.java (hunk 1 above)
Comment (cookie0629) [reply Task_1_4_1/src/main/java/ru/nsu/zhao/FITBook.java:45]: Спасибо за ваше предложение, да, имя переменной должно быть четким и понятным,  examFlag здесь указывает, сдал ли студент итоговый экзамен или нет.

**Pull request merged**